tabular/src/autogluon/tabular/task/tabular_prediction/predictor.py

```diff
@@ -1,1034 +1,389 @@
-        Regression problems do not differ between original and internal representation, and thus this method will return the provided labels.
-        Warning: When `inverse=False`, it is possible for the output to contain NaN label values in multiclass problems if the provided label was dropped during training.
-
-        Parameters
-        ----------
-        labels : `numpy.ndarray` or `pandas.Series`
-            Labels to transform.
-            If `proba=False`, an example input would be the output of `predictor.predict(test_data)`.
-            If `proba=True`, an example input would be the output of `predictor.predict_proba(test_data)`.
-        inverse : boolean, default = False
-            When `True`, the input labels are treated as being in the internal representation and the original representation is outputted.
-        proba : boolean, default = False
-            When `True`, the input labels are treated as probabilities and the output will be the internal representation of probabilities.
-                In this case, it is expected that `labels` be a `pandas.DataFrame` or `numpy.ndarray`.
-                If the `problem_type` is multiclass:
-                    The input column order must be equal to `predictor.class_labels`.
-                    The output column order will be equal to `predictor.class_labels_internal`.
-                    if `inverse=True`, the same logic applies, but with input and output columns interchanged.
-            When `False`, the input labels are treated as actual labels and the output will be the internal representation of the labels.
-                In this case, it is expected that `labels` be a `pandas.Series` or `numpy.ndarray`.
-
-        Returns
-        -------
-        Pandas `pandas.Series` of labels if `proba=False` or Pandas `pandas.DataFrame` of label probabilities if `proba=True`
+    @unpack(set_presets)
+    def fit(self,
+            train_data,
+            tuning_data=None,
+            time_limit=None,
+            presets=None,
+            hyperparameters=None,
+            feature_metadata=None,
+            **kwargs):
+        """
+        Fit models to predict a column of data table based on the other columns.
+
+        # TODO: Move documentation from TabularPrediction.fit to here
+        # TODO: Move num_cpus/num_gpus to ag_args_fit
+        # TODO: consider adding kwarg option for data which has already been preprocessed by feature generator to skip feature generation.
+        # TODO: Remove all `time_limits` in project, replace with `time_limit`
+        # TODO: Add logging for which presets were used
+        # TODO: TabularDataset 'file_path' make so it does not have to be named. Same with 'df'.
+        # TODO: Resolve raw text feature usage in default feature generator
+
+        """
+        if self._learner.is_fit:
+            raise AssertionError('Predictor is already fit! To fit additional models, refer to `predictor.fit_extra`.')
```

Comment:
maybe add for clarity: "refer to predictor.fit_extra, or create a new Predictor"

Reply:
Added